app/instrumentation-tests/src/org/commcare/androidTests/LanguagesTest.kt

```diff
@@ -0,0 +1,135 @@
+package org.commcare.androidTests
+
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.closeSoftKeyboard
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.doesNotExist
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import org.commcare.CommCareApplication
+import org.commcare.dalvik.R
+import org.commcare.utils.CustomMatchers
+import org.commcare.utils.InstrumentationUtility
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class LanguagesTest : BaseTest() {
+
+    companion object {
+        const val CCZ_NAME = "languages.ccz"
+    }
+
+    @Before
+    fun setup() {
+        if (CommCareApplication.instance().currentApp == null) {
+            InstrumentationUtility.installApp(FormEntryTest.CCZ_NAME)
+        } else {
+            InstrumentationUtility.uninstallCurrentApp()
+            InstrumentationUtility.installApp(FormEntryTest.CCZ_NAME)
+            Espresso.pressBack()
+        }
+        InstrumentationUtility.login("user_with_no_data", "123")
+    }
+
+    @Test
+    fun testLanguageChanges() {
+        InstrumentationUtility.openForm(0, 0)
+        onView(withText("Enter a name:"))
+                .check(matches(isDisplayed()))
+        exitForm()
+        InstrumentationUtility.gotoHome()
+        selectLanguageChangeOption()
+        checkLanguageOptions()
+
+        // Confirm the options persist on rotation
+        InstrumentationUtility.rotateLeft()
+        checkLanguageOptions()
+        InstrumentationUtility.rotatePortrait()
+        checkLanguageOptions()
+
+        // Clicking just after rotating doesn't work for some reason, so waiting here for a moment.
+        InstrumentationUtility.sleep(2)
+        onView(withText("Hindi"))
+                .perform(click())
+        onView(withText("Start"))
+                .perform(click())
+        onView(withText("Basic Form Tests"))
```

Comment:
can reuse `InstrumentationUtility.openForm(0, 0)`

Reply:
Strange, I don't know why I did this. Will change it.